src/main/java/org/reactome/release/cosmicupdate/Main.java

```diff
@@ -0,0 +1,227 @@
+			execService.invokeAll(Arrays.asList(unzipper1, unzipper2, unzipper3));
+			execService.shutdown();
+			
+			MySQLAdaptor adaptor = ReleaseStep.getMySQLAdaptorFromProperties(props);
+			loadTestModeFromProperties(props);
+			Collection<GKInstance> cosmicObjects = COSMICUpdateUtil.getCOSMICIdentifiers(adaptor);
+			logger.info("{} COSMIC identifiers", cosmicObjects.size());
+			// Filter the identifiers to exclude the COSV prefixes. 
+			List<GKInstance> filteredCosmicObjects = cosmicObjects.parallelStream().filter(inst -> {
+				try
+				{
+					return !((String)inst.getAttributeValue(ReactomeJavaConstants.identifier)).toUpperCase().startsWith("COSV");
+				}
+				catch (Exception e)
+				{
+					e.printStackTrace();
+				}
+				return false;
+			}).collect(Collectors.toList());
+			logger.info("{} filtered COSMIC identifiers", filteredCosmicObjects.size());
+	
+			Map<String, COSMICIdentifierUpdater> updaters = COSMICUpdateUtil.determinePrefixes(filteredCosmicObjects);
+			
+			COSMICUpdateUtil.validateIdentifiersAgainstFiles(updaters, COSMICFusionExport, COSMICMutationTracking, COSMICMutantExport);
+			COSMICUpdateUtil.printIdentifierUpdateReport(updaters);
+			if (!this.testMode)
+			{
+				updateIdentifiers(adaptor, updaters);
+			}
+		}
+	}
+
+	/**
+	 * Download the data files from COSMIC.
+	 * @throws Exception
+	 */
+	private void downloadFiles() throws Exception
+	{
+		COSMICFileRetriever mutantExportRetriever = new COSMICFileRetriever();
+		COSMICFileRetriever mutationTrackingRetriever = new COSMICFileRetriever();
+		COSMICFileRetriever fusionExportRetriever = new COSMICFileRetriever();
+		
+		try
+		{
+			mutantExportRetriever.setDataURL(new URI(Main.COSMICMutantExportURL));
+			mutationTrackingRetriever.setDataURL(new URI(Main.COSMICMutationTrackingURL));
+			fusionExportRetriever.setDataURL(new URI(Main.COSMICFusionExportURL));
+			
+			final String mutantExportDestination = "./CosmicMutantExport.tsv.gz";
+			final String mutationTrackingDestination = "./CosmicMutationTracking.tsv.gz";
+			final String fusionExportDestination = "./CosmicFusionExport.tsv.gz";
+			
+			mutantExportRetriever.setFetchDestination(mutantExportDestination);
+			mutationTrackingRetriever.setFetchDestination(mutationTrackingDestination);
+			fusionExportRetriever.setFetchDestination(fusionExportDestination);
+			
+			logDownloadMessage(mutantExportRetriever, mutantExportDestination);
+			this.executeDownload(mutantExportRetriever);
+			
+			logDownloadMessage(mutationTrackingRetriever, mutationTrackingDestination);
+			this.executeDownload(mutationTrackingRetriever);
+			
+			logDownloadMessage(fusionExportRetriever, fusionExportDestination);
+			this.executeDownload(fusionExportRetriever);
+		}
+		catch (URISyntaxException e)
+		{
+			e.printStackTrace();
+			throw e;
+		}
+		catch (Exception e)
+		{
+			logger.error("Error occurred while trying to download a file! " + e.getMessage(), e);
```

Comment:
I assume this is the case but will the error message include the name of the file that caused the error?

Reply:
The name of the file being downloaded is logged before the download begins. So if the logs says "downloading file: SomeFile.tgz" and then says "downloading file: SomeOtherFile.tgz" and then says "Error occurred while trying to download a file!", it's a good guess that SomeFile.tgz downloaded OK and SomeOtherFile.tgz didn't download OK.